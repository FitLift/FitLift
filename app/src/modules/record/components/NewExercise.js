import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {
  Text,
} from 'react-native';
import NumberInput from './NumberInput';
import SubmitButton from './SubmitButton';
import {
  HeaderColumn, ExerciseColumn,
} from '../../../components/Header';

const RowStyle = styled.View`
  alignItems: center;
  borderBottomWidth: 1px;
  flexDirection: row;
  height: 70px;
  justifyContent: space-around;
`;

export default class NewExercise extends PureComponent {
  static propTypes = {
    display: PropTypes.bool,
    id: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    reps: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
    submitButtonOnPress: PropTypes.func.isRequired,
    timeStamp: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    weight: PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
    ]),
  };

  static defaultProps = {
    display: false,
    reps: '',
    weight: '',
  };

  onChange = (id, key) => value => this.props.onChange(id, key, value);

  render() {
    const {
      id,
      type,
      reps,
      display,
      submitButtonOnPress,
      timeStamp,
      weight,
    } = this.props;
    return (
      <RowStyle>
        <ExerciseColumn>
          <Text>{type}</Text>
          <Text>{timeStamp}</Text>
        </ExerciseColumn>
        <HeaderColumn>
          <NumberInput
            onChange={this.onChange(id, 'reps')}
            value={reps}
          />
        </HeaderColumn>
        <HeaderColumn>
          <NumberInput
            onChange={this.onChange(id, 'weight')}
            value={weight}
          />
        </HeaderColumn>
        <HeaderColumn>
          <SubmitButton
            id={id}
            display={display}
            onPress={submitButtonOnPress}
          />
        </HeaderColumn>
      </RowStyle>
    );
  }
}
