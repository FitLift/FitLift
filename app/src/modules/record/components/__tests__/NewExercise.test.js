import React from 'react';
import renderer from 'react-test-renderer';
import NewExercise from '../NewExercise';

describe('render tests', () => {
  it('doesnt crash', () => {
    const props = {
      index: 1,
      onChange: f => f,
      reps: 10,
      submitButtonColor: '#9CCC65',
      timeStamp: '7:53:24 pm',
      type: 'bicep curls',
      weight: 10,
    };
    const rendered = renderer.create(
      <NewExercise {...props} />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
