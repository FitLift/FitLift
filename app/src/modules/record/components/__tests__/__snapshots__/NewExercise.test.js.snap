// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render tests doesnt crash 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderBottomWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "height": 70,
      "justifyContent": "space-around",
    }
  }
>
  <View
    style={
      Object {
        "width": 100,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      bicep curls
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      7:53:24 pm
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    reps:
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    maxLength={2}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 4,
        "borderWidth": 1,
        "fontSize": 15,
        "height": 30,
        "textAlign": "center",
        "width": 30,
      }
    }
    value="10"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    weight:
  </Text>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    maxLength={2}
    onChangeText={[Function]}
    style={
      Object {
        "borderColor": "gray",
        "borderRadius": 4,
        "borderWidth": 1,
        "fontSize": 15,
        "height": 30,
        "textAlign": "center",
        "width": 30,
      }
    }
    value="10"
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "backgroundColor": "#9CCC65",
          "borderRadius": 4,
          "height": 30,
          "justifyContent": "center",
          "padding": 5,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
